library/psa_crypto.c

```diff
@@ -1530,6 +1530,36 @@ static psa_status_t psa_export_ffdh_public_key_internal(
     return( status );
 }
 
+static psa_status_t mbedtls_psa_ffdh_generate_key(
+    const psa_key_attributes_t *attributes,
+    uint8_t *key_buffer, size_t key_buffer_size, size_t *key_buffer_length )
+{
+    mbedtls_mpi X, P;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    mbedtls_mpi_init( &P ); mbedtls_mpi_init( &X );
+
+    status = psa_ffdh_set_prime_generator( PSA_BITS_TO_BYTES( attributes->core.bits ), &P, NULL );
+
+    if( status == PSA_SUCCESS )
+    {
+        MBEDTLS_MPI_CHK( mbedtls_mpi_random( &X, 4, &P, mbedtls_psa_get_random,
```

Comment:
I would appreciate some clarification here. According to the [RFC](https://www.ietf.org/rfc/rfc2631.txt):

>   X9.42 requires that the private key x be in the interval [2, (q - 2)]. 

while

> p = jq + 1 where q is a large prime of length m and j>=2.

Since it's the same p as described elsewhere, shouldn't the computations here be a bit different? We're generating a key in the range of `[2, (p - 2)]`, and therefore it can fall well beyond `q-2`.

Reply:
Hmm I'm not sure if you refer to correct RFC. As stated in the issue description https://github.com/Mbed-TLS/mbedtls/issues/3261 the implementation should be based on [RFC7919](https://www.rfc-editor.org/rfc/rfc7919). And there is the following information: 
> Traditional finite field Diffie-Hellman has each peer choose their
   secret exponent from the range [2, p-2]

The same information about generating private key is in this comment: https://github.com/Mbed-TLS/mbedtls/issues/3261#issuecomment-1178666285 from @mpg:
> generate a private key (just an integer in the range [2, p-2])

But I think we should add a comment and refer to RFC7919.